Sources/KlaviyoSwift/Models/ProfileAPIExtension.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  ProfileAPIExtension.swift
+//
+//
+//  Created by Ajay Subramanya on 8/6/24.
+//
+
+import Foundation
+import KlaviyoCore
+
+extension String {
+    fileprivate func returnNilIfEmpty() -> String? {
+        isEmpty ? nil : self
+    }
+}
+
+extension Profile {
+    func toAPIModel(
+        email: String? = nil,
+        phoneNumber: String? = nil,
+        externalId: String? = nil,
+        anonymousId: String) -> ProfilePayload {
+        ProfilePayload(
+            email: email ?? self.email?.returnNilIfEmpty(),
+            phoneNumber: phoneNumber ?? self.phoneNumber?.returnNilIfEmpty(),
+            externalId: externalId ?? self.externalId?.returnNilIfEmpty(),
```

Comment:
Yeah initially I had it the way you are suggesting but it would fail this test, 

```swift 
    func testSetProfileWithEmptyStringIdentifiers() async throws {
        let initialState = KlaviyoState(
            apiKey: TEST_API_KEY,
            email: "foo@bar.com",
            anonymousId: environment.uuid().uuidString,
            phoneNumber: "99999999",
            externalId: "12345",
            pushTokenData: .init(pushToken: "blob_token",
                                 pushEnablement: .authorized,
                                 pushBackground: .available,
                                 deviceData: .init(context: environment.appContextInfo())),
            queue: [],
            requestsInFlight: [],
            initalizationState: .initialized,
            flushing: true)

        let store = TestStore(initialState: initialState, reducer: KlaviyoReducer())

        _ = await store.send(.enqueueProfile(Profile(email: "", phoneNumber: "", externalId: ""))) {
            $0.email = nil // since we reset state
            $0.phoneNumber = nil // since we reset state
            $0.externalId = nil // since we reset state
            $0.enqueueProfileOrTokenRequest()
            $0.pushTokenData = nil
        }
    }
```

I think the test makes sense since when you enqueue a profile (developer identifying a new profile from what is currently in state) we reset out local SDK state and assign whatever the developer is now sending to us but for whatever reason they send an empty string it's going to fail in the API so it's better to convert that to nil than to have make a network call and fail.